src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -516,6 +516,30 @@ def merge_type(a: DataType, b: DataType, name: Optional[str] = None) -> DataType
         return a
 
 
+def python_value_str_to_object(value, tp: DataType) -> Any:
+    if isinstance(tp, StringType):
+        return value
+
+    if isinstance(tp, ArrayType):
+        curr_list = eval(value)
+        if curr_list is None:
+            return None
+        element_tp = tp.element_type or StringType()
+        return [python_value_str_to_object(val, element_tp) for val in curr_list]
+
+    if isinstance(tp, MapType):
+        curr_dict: dict = eval(value)
+        if curr_dict is None:
+            return None
+        key_tp = tp.key_type or StringType()
+        val_tp = tp.value_type or StringType()
+        return {
+            python_value_str_to_object(k, key_tp): python_value_str_to_object(v, val_tp)
+            for k, v in curr_dict.items()
+        }
+    return eval(value)
```

Comment:
what about other types like VariantType, GeometryType? -- it looks like we are only handling primitive types + list + dict here

Reply:
this is a good point. let me look into how we can specify these type-hints